### PR TITLE
drivers: gpio: Convert drivers to new DT device macros

### PR DESCRIPTION
**Pull request by galak**
Convert gpio drivers to use new DT variants of the DEVICE APIs.
    DEVICE_AND_API_INIT -> DEVICE_DT_DEFINE
    DEVICE_GET -> DEVICE_DT_GET
    DEVICE_DECLARE -> DEVICE_DT_INST_DECLARE

etc..

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

**Review by scottwcpg — APPROVED** [review]

File: drivers/gpio/gpio_xlnx_axi.c [hunk 1]
```diff
@@ -281,7 +281,7 @@ static const struct gpio_driver_api gpio_xlnx_axi_driver_api = {
 		.all_outputs = DT_INST_PROP_OR(n, xlnx_all_outputs2, 0),\
 	};								\
 									\
-	DEVICE_AND_API_INIT(gpio_xlnx_axi_##n##_2,			\
+	DEVICE_DT_INST_DEFINE(gpio_xlnx_axi_##n##_2,			\
```
Comment (pabigot) [inline drivers/gpio/gpio_xlnx_axi.c:284]: This doesn't look right.  Is the first argument a node id?  Where's the pm control?  Same below.

**Review by erwango — APPROVED** [review]
+1 for STM32

File: drivers/gpio/gpio_xlnx_axi.c (hunk 1 above)
Comment (pabigot) [reply drivers/gpio/gpio_xlnx_axi.c:284]: NAK to make sure this is addressed.

**Review by pabigot — CHANGES_REQUESTED** [review]

File: drivers/gpio/gpio_xlnx_axi.c (hunk 1 above)
Comment (galak) [reply drivers/gpio/gpio_xlnx_axi.c]: Fixed.

**Review by pabigot — APPROVED** [review]

**Review by franciscomunoz — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**